contrib/views/detail_tabs/curriculum.py

```diff
@@ -61,4 +66,36 @@ def get_context_data(self, **kwargs):
 
         context_data['BE_ISO_CODE'] = BE_ISO_CODE
 
+        # Retrieve more information about the foreign key fields
+        # Store previous data to prevent unnecessary requests
+        retrieved_countries = dict()
+        retrieved_linguistic_regimes = dict()
+
+        for experience in context_data['curriculum_experiences']:
+
+            # Initialize the country
+            if experience.country:
+                if experience.country in retrieved_countries:
+                    experience.country = retrieved_countries.get(experience.country)
+                else:
+                    country = CountriesService.get_country(
+                        iso_code=experience.country,
+                        person=self.request.user.person,
+                    )
+                    experience.country_name = country.name if is_supported_language else country.name_en
+                    retrieved_countries[experience.country] = experience.country_name
+
+            # Initialize the linguistic regime
+            if experience.linguistic_regime:
+                if experience.linguistic_regime in retrieved_linguistic_regimes:
+                    experience.linguistic_regime_name = retrieved_linguistic_regimes.get(experience.linguistic_regime)
+                else:
+                    linguistic_regime = LanguageService.get_language(
+                        code=experience.linguistic_regime,
+                        person=self.request.user.person,
+                    )
+                    experience.linguistic_regime_name = (
+                        linguistic_regime.name if is_supported_language else linguistic_regime.name_en
+                    )
```

Comment:
Peut-être englober ca dans une méthode style initialize_countries_and_linguistic_regimes?

Reply:
Ou en deux méthodes appelées dans le for : initailize_country & initialize_linguistic_regime